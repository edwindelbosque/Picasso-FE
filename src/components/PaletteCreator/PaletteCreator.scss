* {
	box-sizing: border-box;
	padding: 0;
}

body {
	padding: 0;
}

.random-generator__section {
	margin: auto;
	margin-top: 180px;
	display: flex;
	flex-direction: column;
	align-items: center;

	label {
		font-family: 'Roboto';
		color: rgb(77, 77, 77);
	}

	label {
		font-family: 'Roboto';
		color: rgb(77, 77, 77);
	}

	::placeholder {
		color: rgb(207, 207, 207);
	}

	button:disabled {
		opacity: 0.5;
		pointer-events: none;
	}

	input {
		height: 60px;
		padding-left: 1px;
		width: 490px;
		border-bottom: 3px solid rgb(122, 124, 133);

		&:focus {
			outline: none;
		}
	}

	.SavePaletteBtn {
		margin-left: 10px;
		width: 90px;
		background: #1a355c;
		margin: 0 25px;
	}

	button {
		height: 60px;
		width: 170px;
		cursor: pointer;
		border: none;
		background: #2164f3;
		font-size: 18px;
		color: white;
		border-radius: 3px;
		transition: 0.15s;

		&:focus {
			outline: none;
		}

		&:hover {
			background: #3777ff;
			box-shadow: 0 0 15px -6px #3777ff;
		}
	}
}

.random-generator__div-header {
	display: flex;
	justify-content: space-between;
	align-items: flex-end;
}

.palette-name__label-and__input {
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
}

.palette-name__input {
	background: transparent;
	border: none;
	border-bottom: 3px solid black;
	height: 40px;
	font-size: 40px;
	text-align: left;
	font-weight: 700;
}

@media only screen and (max-width: 835px) {
	body {
		padding: 0;
	}

	.random-generator__section {
		label {
			font-family: 'Roboto';
			color: rgb(77, 77, 77);
		}

		label {
			font-family: 'Roboto';
			color: rgb(77, 77, 77);
		}

		::placeholder {
			color: rgb(207, 207, 207);
		}

		button:disabled {
			opacity: 0.5;
			pointer-events: none;
		}

		input {
			height: 60px;
			padding-left: 1px;
			width: 300px;
			border-bottom: 3px solid rgb(122, 124, 133);
			font-size: 30px;
			padding-top: 15px;

			&:focus {
				outline: none;
			}
		}

		.SavePaletteBtn {
			width: 90px;
			background: #1a355c;
		}

		button {
			height: 60px;
			width: 170px;
			cursor: pointer;
			border: none;
			background: #2164f3;
			font-size: 18px;
			color: white;
			border-radius: 3px;
			transition: 0.15s;

			&:focus {
				outline: none;
			}

			&:hover {
				background: #3777ff;
				box-shadow: 0 0 15px -6px #3777ff;
			}
		}
	}

	.random-generator__div-header {
		display: flex;
		justify-content: space-between;
		align-items: flex-end;
	}

	.palette-name__label-and__input {
		display: flex;
		flex-direction: column;
		justify-content: flex-start;
	}

	.palette-name__input {
		background: transparent;
		border: none;
		border-bottom: 3px solid black;
		height: 40px;
		font-size: 40px;
		text-align: left;
		font-weight: 700;
	}
}
