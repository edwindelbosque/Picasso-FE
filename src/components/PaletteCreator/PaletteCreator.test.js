import React from 'react';
import PaletteCreator from './PaletteCreator';
import { shallow } from 'enzyme';
jest.mock('react-redux', () => ({
	useDispatch: () => {},
	useSelector: () => ({ catalogs: [{}, {}] })
}));

describe('PaletteCreator', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<PaletteCreator />);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
