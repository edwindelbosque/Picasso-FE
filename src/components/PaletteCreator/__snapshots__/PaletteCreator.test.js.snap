// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCreator App should match snapshot 1`] = `
<Fragment>
  <section
    className="random-generator__section"
  >
    <div
      className="random-generator__div-header"
    >
      <div
        className="palette-name__label-and__input"
      >
        <input
          className="palette-name__input"
          id="palette-name"
          name="palette-name"
          onChange={[Function]}
          placeholder="Palette Name"
          type="text"
          value=""
        />
      </div>
      <button
        className="SavePaletteBtn"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
      <button
        className="GenerateColorsBtn"
        onClick={[Function]}
        type="button"
      >
        Generate Colors
      </button>
    </div>
    <Colors
      lockedColors={
        Object {
          "catalogs": Array [
            Object {},
            Object {},
          ],
        }
      }
      palette={
        Object {
          "catalogs": Array [
            Object {},
            Object {},
          ],
        }
      }
    />
  </section>
  <SaveMenu
    catalogs={
      Object {
        "catalogs": Array [
          Object {},
          Object {},
        ],
      }
    }
    postPalette={[Function]}
  />
</Fragment>
`;
